lib/faraday/adapter.rb

```diff
@@ -65,5 +65,31 @@ def save_response(env, status, body, headers = nil, reason_phrase = nil)
       env.response.finish(env) unless env.parallel?
       env.response
     end
+
+    # Fetches either a read, write, or open timeout setting. Defaults to the
+    # :timeout value if a more specific one is not given.
+    #
+    # @param type [Symbol] Describes which timeout setting to get: :read,
+    #                      :write, or :open.
+    # @param options [Hash] Hash containing Symbol keys like :timeout,
+    #                       :read_timeout, :write_timeout, :open_timeout, or
+    #                       :timeout
+    #
+    # @return [Integer, nil] Timeout duration in seconds, or nil if no timeout
+    #                        has been set.
+    def request_timeout(type, options)
+      key = TIMEOUT_KEYS[type]
```

Comment:
Could use a `key = TIMEOUT_KEYS.fetch(type) do ... end` and raise the error in there?

Reply:
(We can do that in a follow-up PR, no biggie.)